### ISSUE TITLE
Question about result of DyNeRF dataset : a difference between the results in the paper and my results

### ISSUE THREAD

**Issue by minsu1206** [sarafridov/K-Planes#30]
Hi,

Thank you for sharing your work and the interesting model you've developed. 
I hope that your model becomes the new standard for Dynamic NeRF models.

I trained the K-planes hybrid version using the DyNeRF dataset, following your instructions without making any other changes to the configuration. Here's what I did:

1. Precomputed IST & ISG weights by setting downsample=4, global_step=1.
2. Trained the model by setting downsample=2, global_step=90001 back again.

For training, I used one A100 32GB and torch=1.10.1.

I compared my results with the results from the supplementary paper, and I marked my results in red. 

 ![image](https://github.com/sarafridov/K-Planes/assets/59243570/0afa9772-0305-4acd-9bc7-d8611afaaccb)

There seems to be a mismatch in the metrics, and the performance appears to be worse than MixVoxel-L. 
My question is, how can I achieve the same results as described in the paper?

Best regards.

**Comment by sarafridov** [comment]
Hmm I'm not sure what could be going on here since I am able to reproduce the numbers in the paper (at least I checked for the salmon scene), but some things you can try are: (1) in case there's any issue with the preprocessing for the ray sampling weights, you can try using some that I uploaded for a few of the scenes (the .pt files [here](https://drive.google.com/drive/u/1/folders/1IpylGEiImGJbh7CL7ij2xCdrCnmjD4Sj) for salmon and in flamesteak_explicit and searsteak_explicit). If you download these and put them in your data directory, then you can go straight to the full training with the default config and it should work. (2) I doubt this would be the issue, but you can double check that your python environment matches mine in case this is due to some package mismatch; I uploaded my environment package list with the version info [here](https://drive.google.com/drive/u/1/folders/1zs_folzaCdv88y065wc6365uSRfsqITH). Note that this includes some unnecessary packages along with the ones that are actually used (sorry about that, we might upload a cleaner version later).

**Comment by xuyaojian123** [comment]
> Hmm I'm not sure what could be going on here since I am able to reproduce the numbers in the paper (at least I checked for the salmon scene), but some things you can try are: (1) in case there's any issue with the preprocessing for the ray sampling weights, you can try using some that I uploaded for a few of the scenes (the .pt files [here](https://drive.google.com/drive/u/1/folders/1IpylGEiImGJbh7CL7ij2xCdrCnmjD4Sj) for salmon and in flamesteak_explicit and searsteak_explicit). If you download these and put them in your data directory, then you can go straight to the full training with the default config and it should work. (2) I doubt this would be the issue, but you can double check that your python environment matches mine in case this is due to some package mismatch; I uploaded my environment package list with the version info [here](https://drive.google.com/drive/u/1/folders/1zs_folzaCdv88y065wc6365uSRfsqITH). Note that this includes some unnecessary packages along with the ones that are actually used (sorry about that, we might upload a cleaner version later).

Could you upload DyNeRF full scenes hybrid pretrained models? Thanks a lot!

**Comment by xuyaojian123** [comment]
> Hi,
> 
> Thank you for sharing your work and the interesting model you've developed. I hope that your model becomes the new standard for Dynamic NeRF models.
> 
> I trained the K-planes hybrid version using the DyNeRF dataset, following your instructions without making any other changes to the configuration. Here's what I did:
> 
> 1. Precomputed IST & ISG weights by setting downsample=4, global_step=1.
> 2. Trained the model by setting downsample=2, global_step=90001 back again.
> 
> For training, I used one A100 32GB and torch=1.10.1.
> 
> Then, I evaluated models with --validate-only and --log-dir /path/to/dir_of_model
> 
> I compared my results with the results from the supplementary paper, and I marked my results in red.
> 
> ![image](https://private-user-images.githubusercontent.com/59243570/252662385-0afa9772-0305-4acd-9bc7-d8611afaaccb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1ODY3NDgsIm5iZiI6MTcwNTU4NjQ0OCwicGF0aCI6Ii81OTI0MzU3MC8yNTI2NjIzODUtMGFmYTk3NzItMDMwNS00YWNkLTliYzctZDg2MTFhZmFhY2NiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDE0MDA0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMzU3ZTNmNjEyNmJjOWVhY2JmODZhYjM0NjRkMjA3NjBlMzlkMDM3MDU2ZDcxODcxMWRlNTU2YjM4MWViOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jFz0IMFLNf57jPr_7j3tRld9sxYYqzXT4F92wXylWSo)
> 
> There seems to be a mismatch in the metrics, and the performance appears to be worse than MixVoxel-L. My question is, how can I achieve the same results as described in the paper?
> 
> Best regards.

May I ask if you got the correct result later? My results are even a little worse than yours, especially the SSIM metric.

**Comment by minsu1206** [comment]
Sorry, I tried 2~3 times again, but couldn't get it.

**Comment by xuyaojian123** [comment]
I used your pre-trained model to get metrics, but  also differently from the paper metrics (testdata downsamle×2), especially the ssim metrics. Is there any difference in how ssim metrics are calculated?

**Comment by kapil-4944** [comment]
@ minsu1206 The results compared in the SSIM metric are similar to the values given in the MS-SSIM metric in pre-trained results shared by the authors, so they must have compared with the same metric in all the other models as well.
@sarafridov, is this the case? 
By the way, the MS-SSIM metric should be a more reliable metric for videos.
